src/mpid/ch4/shm/posix/release_gather/nb_bcast_release_gather.h

```diff
@@ -488,9 +489,11 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_nb_release_gather_ibcast_impl(void *loc
         if (my_rank != root) {
             /* Non-root unpack the data if expecting non-contiguous datatypes */
             /* Dependency is created from the last vertices of each chunk to the unpack vertex */
+            MPIR_Assert(num_chunks <= INT_MAX);
             mpi_errno =
                 MPIR_TSP_sched_localcopy(local_buf, nbytes, MPI_BYTE, ori_local_buf, ori_count,
-                                         ori_datatype, sched, num_chunks, sixth_vtx_id, &vtx_id);
+                                         ori_datatype, sched, (int) num_chunks, sixth_vtx_id,
```

Comment:
I think we should write a safe typecast that aborts on overflow, instead of just do type cast here. It would at least complains if the overflow happens.

Reply:
Something like
```c
int Aint2int(MPI_Aint val)
{
    if (val > INT_MAX || val < INT_MIN) {
        MPIR_Assert(0);
    }
    return (int) val;
}
```
Unfortunately default `assert()` won't work here.

Reply:
This macro works
```c
#define TO_INT(val) \
    ((val) > INT_MAX || (val) < INT_MIN ? (assert(0), 0) : (int) (val))
```

Reply:
I always liked this old trick from Dave Goodell: maybe ten years later we can try it again  10ecd01